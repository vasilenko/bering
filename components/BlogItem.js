import React from 'react';
import PropTypes from 'prop-types';

import Image from './Image';
import TextBox from './TextBox';

const BlogItem = ({ post }) => (
  <div>
    <Image {...post.image} />
    <TextBox>{post.text}</TextBox>
  </div>
);

BlogItem.propTypes = {
  post: PropTypes.shape({
    text: PropTypes.string
    image: PropTypes.shape(Image.propTypes),
  })
};

export default BlogItem;
