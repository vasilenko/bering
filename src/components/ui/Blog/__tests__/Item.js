import React from 'react';
import { shallow } from 'enzyme';

import Link from 'components/ui/Link';

import BlogItem from '../Item';

const post = {
  id: 0,
  title: 'Title',
  text: 'Text',
  image: {
    src: 'image.png'
  },
  meta: {
    createdAt: '9/5/2017, 4:34:58 PM'
  }
};

const item = shallow(<BlogItem post={post} />);

describe('BlogItem', () => {
  it('renders title', () => {
    const header = <Link to={`/posts/${post.id}`}>{post.title}</Link>;

    expect(item.contains(header)).toEqual(true);
  });

  it('matches snapshot', () => {
    expect(item).toMatchSnapshot();
  });
});
