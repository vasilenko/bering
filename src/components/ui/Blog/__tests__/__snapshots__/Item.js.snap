// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <ItemImage
      size="medium"
    >
      <Image
        src="image.png"
      />
    </ItemImage>
    <ItemContent>
      <ItemHeader>
        <Link
          replace={false}
          to="/posts/0"
        >
          Title
        </Link>
      </ItemHeader>
      <ItemDescription>
        <TextBox>
          Text
        </TextBox>
      </ItemDescription>
      <ItemExtra>
        <BlogMeta
          author="Anonymous"
          createdAt="9/5/2017, 4:34:58 PM"
          postId={0}
        />
      </ItemExtra>
    </ItemContent>
  </Item>,
  "nodes": Array [
    <Item>
      <ItemImage
        size="medium"
      >
        <Image
          src="image.png"
        />
      </ItemImage>
      <ItemContent>
        <ItemHeader>
          <Link
            replace={false}
            to="/posts/0"
          >
            Title
          </Link>
        </ItemHeader>
        <ItemDescription>
          <TextBox>
            Text
          </TextBox>
        </ItemDescription>
        <ItemExtra>
          <BlogMeta
            author="Anonymous"
            createdAt="9/5/2017, 4:34:58 PM"
            postId={0}
          />
        </ItemExtra>
      </ItemContent>
    </Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        post={
          Object {
            "id": 0,
            "image": Object {
              "src": "image.png",
            },
            "meta": Object {
              "createdAt": "9/5/2017, 4:34:58 PM",
              "postId": 0,
            },
            "text": "Text",
            "title": "Title",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "post": Object {
            "id": 0,
            "image": Object {
              "src": "image.png",
            },
            "meta": Object {
              "createdAt": "9/5/2017, 4:34:58 PM",
              "postId": 0,
            },
            "text": "Text",
            "title": "Title",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <ItemImage
            size="medium"
          >
            <Image
              src="image.png"
            />
          </ItemImage>
          <ItemContent>
            <ItemHeader>
              <Link
                replace={false}
                to="/posts/0"
              >
                Title
              </Link>
            </ItemHeader>
            <ItemDescription>
              <TextBox>
                Text
              </TextBox>
            </ItemDescription>
            <ItemExtra>
              <BlogMeta
                author="Anonymous"
                createdAt="9/5/2017, 4:34:58 PM"
                postId={0}
              />
            </ItemExtra>
          </ItemContent>
        </Item>,
        "_debugID": 2,
        "_renderedOutput": <Item>
          <ItemImage
            size="medium"
          >
            <Image
              src="image.png"
            />
          </ItemImage>
          <ItemContent>
            <ItemHeader>
              <Link
                replace={false}
                to="/posts/0"
              >
                Title
              </Link>
            </ItemHeader>
            <ItemDescription>
              <TextBox>
                Text
              </TextBox>
            </ItemDescription>
            <ItemExtra>
              <BlogMeta
                author="Anonymous"
                createdAt="9/5/2017, 4:34:58 PM"
                postId={0}
              />
            </ItemExtra>
          </ItemContent>
        </Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    post={
      Object {
        "id": 0,
        "image": Object {
          "src": "image.png",
        },
        "meta": Object {
          "createdAt": "9/5/2017, 4:34:58 PM",
          "postId": 0,
        },
        "text": "Text",
        "title": "Title",
      }
    }
  />,
}
`;
